### DISCUSSION TITLE
flutter_gen failing if unit tests are running

### DISCUSSION THREAD

**Discussion by chrassendk** [codemagic-ci-cd/codemagic-docs#1745] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Using `flutter_localization 0.1.11` in pubspec.yaml
2. Adding `flutter test` to codemagic.yaml:
```

 - name: Unit tests
           script: | 
            flutter test

```

2. When codemagic reaches "Flutter build ipa and automatic versioning" the following error is thrown:

```
Running pod install...                                              3.2s
Running Xcode build...                                          
Xcode archive done.                                         62.5s
Failed to build iOS app
Error (Xcode): Error: Couldn't resolve the package 'flutter_gen' in 'package:flutter_gen/gen_l10n/app_localizations.dart'.
Encountered error while archiving for device.
Build failed :|
Step 9 script `Flutter build ipa and automatic versioning` exited with status code 1

```

If I disable unit test block it works fine.

### Expected results

It should pass

### Actual results

It fails

### Build id (optional)

648069c84f1a6e6689e26fd4

**Comment by himesh-cm** [comment]
Hi @chrassendk, Thanks for sharing the information. Can you please provide the build id as well?

**Reply by chrassendk** [reply]
Thanks, sorry about that. It is `648069c84f1a6e6689e26fd4`

**Reply by himesh-cm** [reply]
@chrassendk, Can you please run another build on `flutter:stable channel`, along with `unit tests`?
Also Please confirm that `flutter_gen` package is included as a dev dependency in your `pubspec.yaml` file.

**Reply by chrassendk** [reply]
@himesh-cm latest build `649091db85548cd0bafd8d5f` still failing with same error. Running stable channel and re-added unit tests. The `flutter_gen` can not be part of the dev dependency afaik. see https://docs.flutter.dev/accessibility-and-localization/internationalization and https://github.com/flutter/website/blob/main/examples/internationalization/gen_l10n_example/pubspec.yaml#L16

**Reply by chrassendk** [reply]
@himesh-cm Any additional information I can provide?

**Reply by himesh-cm** [reply]
All the configuration looks alright, 
i also caught an additional error in `pod install`

```
[!] CocoaPods could not find compatible versions for pod "Firebase/CoreOnly":
  In snapshot (Podfile.lock):
    Firebase/CoreOnly (= 10.7.0, ~> 10.7.0)

  In Podfile:
    firebase_core (from `Flutter/ephemeral/.symlinks/plugins/firebase_core/macos`) was resolved to 2.13.1, which depends on
      Firebase/CoreOnly (~> 10.9.0)
 ```     
I suggest you fix this one first, to be sure.

Navigate to your iOS folder in your Flutter project by using a terminal, usually with 
`cd ios`.
Then, delete Podfile.lock by running 
`rm Podfile.lock`.

You can do it locally and re-run the build on codemagic.

Also, then, if you are still facing `flutter_gen` issue, please add the command `flutter clean`, before running the `flutter packages pub get`. Let us know how it goes.

**Reply by chrassendk** [reply]
@himesh-cm Thanks for an elaborate answer, but unfortunately still the same. Build `6496a15ffe3fdd50b48e417f` fails in `Flutter build ipa and automatic versioning` with 
```
Running pod install...                                           2,858ms
Running Xcode build...                                          
Xcode archive done.                                         52.1s
Failed to build iOS app
Error (Xcode): Error: Couldn't resolve the package 'flutter_gen' in 'package:flutter_gen/gen_l10n/app_localizations.dart'.
```

**Reply by himesh-cm** [reply]
Hi @chrassendk, I did a quick test, and i was able to run `unit tests` and `code signing` with `flutter localization` enabled without any errors.
What version of `flutter_localization` package are you using, it is latest?

**Reply by chrassendk** [reply]
Im using the latest and it is still failing under build id `649af3d213347a101f537bbf`

**Reply by chrassendk** [reply]
@himesh-cm I have this mentioned in my pubspec:

```
dependencies:
  flutter_localization: ^0.1.12
  flutter_localizations:
    sdk: flutter
  ....  
```
nothing in my dev_dependencies:
```
dev_dependencies:
  flutter_lints: ^2.0.0
  json_serializable: ^6.6.1
  build_runner: ^2.3.3
  freezed: ^2.3.2
  flutter_test:
    sdk: flutter
  flutter_launcher_icons: "^0.13.1"
```

**Reply by himesh-cm** [reply]
@chrassendk, Can you please confirm if you have steps 2, 3 and 4 setup from the docs as shown [here](https://docs.flutter.dev/accessibility-and-localization/internationalization#adding-your-own-localized-messages).

**Reply by chrassendk** [reply]
@himesh-cm That is confirmed. They are configured correctly

**Reply by himesh-cm** [reply]
@chrassendk Thanks for the reply; l am out of ideas here. Let me check with the team.

**Reply by himesh-cm** [reply]
Hi @chrassendk, i was going through the `flutter_gen` pub.dev package [here](https://pub.dev/packages/flutter_gen). Are you using this for asset generation, fonts, etc?
Can you share more light on the usage?

**Reply by chrassendk** [reply]
@himesh-cm It is pretty straightforward the usecase. Im only trying to localize strings following this: https://docs.flutter.dev/accessibility-and-localization/internationalization

**Reply by himesh-cm** [reply, marked as answer]
Hi @chrassendk,
Can you please rearrange the script in the below order and share the build id?
1. Installing Flutter Dependencies
2. Install pods
3. Unit Tests
4. Setup `xcode-project use-profiles`
5. flutter ipa command

**Reply by chrassendk** [reply]
@himesh-cm You deserve a star! That worked, sorry for the hassle and thanks.
(cc @mikhail-tokarev )

**Reply by himesh-cm** [reply]
@chrassendk Happy to help. I am glad it is resolved.
While I’m closing the discussion for now, if you have any other questions or suggestions, feel free to reply or start a new thread anytime. Have a nice day!

**Comment by mikhail-tokarev** [comment]
bump

**Reply by chrassendk** [reply]
@mikhail-tokarev @himesh-cm any news on this issue?
